hnn_core/parallel_backends.py

```diff
@@ -0,0 +1,265 @@
+"""Parallel backends"""
+
+# Authors: Blake Caldwell <blake_caldwell@brown.edu>
+#          Mainak Jas <mainakjas@gmail.com>
+
+from warnings import warn
+
+_backend = None
+_loaded_dll = None
+
+
+class Joblib_backend(object):
+    """The Joblib_backend class.
+
+    Parameters
+    ----------
+    n_jobs : int
+        The number of jobs to start in parallel
+
+    Attributes
+    ----------
+    n_jobs : int
+        The number of jobs to start in parallel
+    """
+    def __init__(self, n_jobs=1):
+        self.type = 'joblib'
+
+        self.n_jobs = n_jobs
+        print("joblib will run over %d jobs" % (self.n_jobs))
+
+    def _parallel_func(self, func):
+        if self.n_jobs != 1:
+            try:
+                from joblib import Parallel, delayed
+            except ImportError:
+                warn('joblib not installed. Cannot run in parallel.')
+                self.n_jobs = 1
+        if self.n_jobs == 1:
+            my_func = func
+            parallel = list
+        else:
+            parallel = Parallel(self.n_jobs)
+            my_func = delayed(func)
+
+        return parallel, my_func
+
+    def __enter__(self):
+        global _backend
```

Comment:
I'd declare all the global variables at the top of the script and in capital letters so they are all in one place

Reply:
see how we do it in MNE: https://github.com/mne-tools/mne-python/blob/95ad0eec8a2e1e9c7c74e25852157e650aea7df0/mne/viz/backends/renderer.py#L16-L17

Reply:
Fixed using the MNE convention with caps.